jmclient/jmclient/schedule.py

```diff
@@ -97,38 +108,69 @@ def lower_bounded_int(thelist, lowerbound):
                                options['txcountparams'][1], options['mixdepthcount'])
     txcounts = lower_bounded_int(txcounts, options['mintxcount'])
     tx_list = []
+    ### stage 1 coinjoins, which sweep the entire mixdepth without creating change
+    highest_initial_filled_mixdepth = 0
+    for mixdepth, balance in mixdepth_balance_dict.items():
+        if balance > 0:
+            highest_initial_filled_mixdepth = mixdepth
+    stage1_tx_count = highest_initial_filled_mixdepth + 1
```

Comment:
What happens if mixdepth 0 and mixdepth 2 contain coins, but mixdepth 1 doesn't?
Won't this (section of) code end up scheduling a sweep from mixdepth 1?

Reply:
Well spotted. I'll recode.